packages/@aws-cdk/core/lib/resource.ts

```diff
@@ -1,5 +1,6 @@
+import { Construct } from 'constructs';
 import { ArnComponents } from './arn';
-import { Construct, IConstruct } from './construct-compat';
+import { IConstruct, Construct as cdkConstruct } from './construct-compat';
```

Comment:
Rename to `CoreConstruct` or `ConstructCompat`. At any rate, capital first latter